presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformCorrelatedScalarAggregationToJoin.java

```diff
@@ -38,11 +38,15 @@
 import com.facebook.presto.sql.planner.plan.PlanNode;
```

Comment:
I didn't realize that, but you are right.  There were two problems 
1. I put the new rewrite before PredicatePushDown, so it hadn't been called
2. The predicate push down rewrite doesn't split out the join criteria when the join expression isn't new (I guess it wasn't thinking about joins generated during optimization). 

I'll revert this commit and fix the PredicatePushdown rewrite instead